packages/neuron-ui/src/exceptions/index.ts

```diff
@@ -110,3 +110,21 @@ export class ValueReservedException extends Error {
     this.i18n = { value }
   }
 }
+
+export type ErrorWithI18n = {
+  message: string
+  code: ErrorCode
+  i18n?: Record<string, string>
+}
+
+export function isErrorWithI18n(error: unknown): error is ErrorWithI18n {
```

Comment:
`unknown` is for outside users, `any` could be used there and return `true` if `typeof error?.message === 'string' && typeof error.code === 'number'`